### PR TITLE
Basic requirements VPN GW

### PR DESCRIPTION
**Pull request by monty16597**
# Fix
- `CreateVpnGateway` and `DescribeVpnGateway` responses
- API `DetachVpnGateway`

# Terraform Test cases
- `TestAccAWSVpnGateway_basic`
- `TestAccAWSVpnGateway_delete`
- `TestAccAWSVpnGateway_tags`
- `TestAccAWSVpnGatewayAttachment`

### CODE REVIEW

**Comment by monty16597** [comment]
@bblommers We are ready to merge this PR.

File: moto/ec2/models.py [hunk 1]
```diff
@@ -6675,7 +6675,8 @@ def delete_vpn_gateway(self, vpn_gateway_id):
     def detach_vpn_gateway(self, vpn_gateway_id, vpc_id):
         vpn_gateway = self.get_vpn_gateway(vpn_gateway_id)
         self.get_vpc(vpc_id)
-        detached = vpn_gateway.attachments.pop(vpc_id, None)
+        detached = vpn_gateway.attachments.get(vpc_id, None)
+        detached.state = "detached"
```
Comment (bblommers) [inline moto/ec2/models.py:6679]: The state should be assigned after the None-check (`if not detached`) - otherwise we'll get a uncaught exception

**Review by bblommers — CHANGES_REQUESTED** [review]

File: moto/ec2/models.py (hunk 1 above)
Comment (monty16597) [reply moto/ec2/models.py:6679]: Make sense

**Review by bblommers — APPROVED** [review]
LGTM! Thanks for the fix @monty16597

**Pull request merged**